app/volume.go

```diff
@@ -169,15 +169,58 @@ func NewJob(managerURL, volumeName, snapshotName string, labels map[string]strin
 	}, nil
 }
 
-func (job *Job) run() (err error) {
+// handleVolumeDetachment decides whether the current recurring job should detach the volume.
+// It should detach the volume when:
+// 1. The volume is attached by this recurring job
+// 2. The volume state is VolumeStateAttached
+func (job *Job) handleVolumeDetachment() {
 	volumeAPI := job.api.Volume
 	volumeName := job.volumeName
+	jobName, _ := job.labels[types.RecurringJobLabel]
+	if jobName == "" {
+		logrus.Warn("Missing RecurringJob label")
+		return
+	}
+
+	logrus.Infof("Handling volume detachment for volume %v", volumeName)
+	for {
+		volume, err := volumeAPI.ById(volumeName)
```

Comment:
What if the volume just becomes state `Detached` during the auto reattachment and when this line is executed?

Reply:
Awesome catch! You are right! I will fix this.

Reply:
Fixed